### PR TITLE
Clean up metadata and clarify what small changes affect

### PR DESCRIPTION
**Pull request by ChrisChinchilla**
### Checklist
- [x] All tests are passing
- [x] README / documentation was extended, if necessary
- [x] Used meaningful commit messages

### Description
Closes https://github.com/ethereum/solidity/issues/1644.

Also took the opportunity to tidy up surrounding explanations, let me know if it's still not clear enough, I didn't want to belate the point, but maybe it could be clearer.

### CODE REVIEW

File: docs/metadata.rst [hunk 1]
```diff
@@ -4,28 +4,28 @@ Contract Metadata
...
 data provider.
 
-Of course, you have to publish the metadata file to Swarm (or some other service)
-so that others can access it. The file can be output by using ``solc --metadata``
-and the file will be called ``ContractName_meta.json``.
-It will contain Swarm references to the source code, so you have to upload
-all source files and the metadata file.
+You have to publish the metadata file to Swarm (or another service) so that
+others can access it. You create the file by using the  ``solc --metadata``
+command that creates a file called ``ContractName_meta.json``. It contains
```
Comment (chriseth) [inline docs/metadata.rst]: Repetition of `create`

File: docs/metadata.rst [hunk 2]
```diff
@@ -96,11 +96,11 @@ Comments are of course also not permitted and used here only for explanatory pur
...
     Note the ABI definition above has no fixed order. It can change with compiler versions.
 
-.. note::
-    Since the bytecode of the resulting contract contains the metadata hash, any change to
-    the metadata will result in a change of the bytecode. Furthermore, since the metadata
-    includes a hash of all the sources used, a single whitespace change in any of the source
-    codes will result in a different metadata, and subsequently a different bytecode.
+Since the bytecode of the resulting contract contains the metadata hash, any
+change to the metadata results in a change of the bytecode. This includes
+changes to filename or path, and since the metadata includes a hash of all the
```
Comment (chriseth) [inline docs/metadata.rst]: filenames or paths?

Comment (ChrisChinchilla) [reply docs/metadata.rst]: I took it as being singular and changed accordingly. Let me know if I am wrong.

**Review by chriseth — DISMISSED** [review]
Please squash.

**Comment by ChrisChinchilla** [comment]
@chriseth Squashed.

**Review by chriseth — APPROVED** [review]

File: docs/metadata.rst [hunk 3]
```diff
@@ -96,11 +96,11 @@ Comments are of course also not permitted and used here only for explanatory pur
 .. note::
     Note the ABI definition above has no fixed order. It can change with compiler versions.
 
-.. note::
-    Since the bytecode of the resulting contract contains the metadata hash, any change to
-    the metadata will result in a change of the bytecode. Furthermore, since the metadata
-    includes a hash of all the sources used, a single whitespace change in any of the source
-    codes will result in a different metadata, and subsequently a different bytecode.
+Since the bytecode of the resulting contract contains the metadata hash, any
```
Comment (axic) [inline docs/metadata.rst:99]: Why is this better as not a note?

Comment (chriseth) [reply docs/metadata.rst:99]: Yeah, should be undone.

Comment (ChrisChinchilla) [reply docs/metadata.rst:99]: I have no idea what my reasoning here was now, but it's been merged already :/ I'll create another PR.

**Pull request merged**